pkg/storage/internal/actions/process.go

```
@@ -369,7 +369,16 @@ func (p *process) broadcastTxs(ctx context.Context, txIDs []string, isDelayed bo
 
 	logger.DebugContext(ctx, "Categorizing transactions by status")
 
-	for txID, currentStatus := range knownTxStatusesLookup {
+	for _, txID := range txIDs {
+		currentStatus, ok := knownTxStatusesLookup[txID]
```

Many thanks for finding and fixing a place where we wrongly changed the order of provided `txIDs`.

I think, we don't have to worry about `!ok` case - because it should not happen. Check how `getKnownTxStatuses` works.

So, instead of `DebugLog` and proceeding I would return an error.